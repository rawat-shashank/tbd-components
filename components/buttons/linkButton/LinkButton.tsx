import React from 'react';
import { StyledLinkButton } from './LinkButton.styles';
import { LinkButtonProps } from './LinkButton.types';

/**
 * Display a button as link with link styles.
 */
export const LinkButton = ({ label, className, disabled }: LinkButtonProps) => (
  <StyledLinkButton
    className={className}
    data-testid="link-button"
    disabled={disabled}
  >
    {label}
  </StyledLinkButton>
);
