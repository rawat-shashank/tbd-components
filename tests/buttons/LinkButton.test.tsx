import React from 'react';
import { render } from '@testing-library/react';

import { Default, Disabled } from '../../stories/buttons/LinkButton.stories';
import { ThemeProvider } from 'styled-components';

describe('Unit Test | Link Button Component', () => {
  it('should match label', () => {
    const renderComponent = () => render(<Default {...Default.args} />);
    const { getByTestId } = renderComponent();
    const testComponent = getByTestId('link-button');
    expect(testComponent).toHaveStyle({ color: '#485BFF' });
    expect(testComponent).toHaveTextContent(Default.args.label);
  });

  it('should have attribute disable', () => {
    const renderComponent = () => render(<Disabled {...Disabled.args} />);
    const { getByTestId } = renderComponent();
    const testComponent = getByTestId('link-button');
    expect(testComponent).toHaveStyle({ color: '#A0A0A0' });
    expect(testComponent).toBeDisabled();
  });

  it('should override with theme color', () => {
    const theme: any = { text: { info: 'red' } };
    const renderComponent = () =>
      render(
        <ThemeProvider theme={theme}>
          <Default {...Default.args} />
        </ThemeProvider>
      );
    const { getByTestId } = renderComponent();
    const testComponent = getByTestId('link-button');
    expect(testComponent).toHaveStyle({ color: theme.text.info });
  });

  it('should override with theme disabled color', () => {
    const theme: any = { icon: { disabled: 'black' } };
    const renderComponent = () =>
      render(
        <ThemeProvider theme={theme}>
          <Disabled {...Disabled.args} />
        </ThemeProvider>
      );
    const { getByTestId } = renderComponent();
    const testComponent = getByTestId('link-button');
    expect(testComponent).toHaveStyle({ color: theme.icon.disabled });
  });

  it('should work with external className', () => {
    const renderComponent = () =>
      render(<Default {...Default.args} className="test" />);
    const { getByTestId } = renderComponent();
    const testComponent = getByTestId('link-button');
    expect(testComponent).toHaveClass('test');
  });

  it('should work with external style', () => {
    const style = {
      color: 'red',
    };
    const renderComponent = () =>
      render(<Default {...Default.args} style={style} />);
    const { getByTestId } = renderComponent();
    const testComponent = getByTestId('link-button');
    expect(testComponent).toHaveStyle({ color: style.color });
  });

  it('should match snapshot', () => {
    const { container } = render(<Default {...Default.args} />);
    expect(container).toMatchSnapshot();
  });
});
