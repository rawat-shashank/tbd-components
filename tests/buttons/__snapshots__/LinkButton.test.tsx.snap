// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test | Link Button Component should match snapshot 1`] = `
<div>
  <button
    class="sc-bdvvtL IbS"
    data-testid="link-button"
  >
    Link Button
  </button>
</div>
`;
